File: dailyQuestion/2020/2020-10/10-06/python/solution_one_point.py
def sortColors(nums: [int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    n = len(nums)
    ptr = 0
    for i in range(n) :
        if nums[i] == 0 :
            nums[i] , nums[ptr] = nums[ptr] , nums[i]
            ptr += 1
    for i in range(n) :
        if nums[i] == 1 :
            nums[i], nums[ptr] = nums[ptr] , nums[i]
            ptr += 1

if __name__ == "__main__" :
    nums = [2,0,2,1,1,0]
    sortColors(nums)
    print(nums)

File: dailyQuestion/2020/2020-10/10-06/golang/solution_one_point.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{2, 0, 2, 1, 1, 0}
	sortColors(nums)
	fmt.Println(nums)
}

func sortColors(nums []int) {
	n := len(nums)
	ptr := 0
	for i := 0; i < n; i++ {
		if nums[i] == 0 {
			nums[i], nums[ptr] = nums[ptr], nums[i]
			ptr++
		}
	}
	for i := 0; i < n; i++ {
		if nums[i] == 1 {
			nums[i], nums[ptr] = nums[ptr], nums[i]
			ptr++
		}
	}
}
